core/chains/evm/logpoller/log_poller.go

```diff
@@ -130,6 +156,7 @@ func NewLogPoller(orm *ORM, ec Client, lggr logger.Logger, pollPeriod time.Durat
 		replayComplete:    make(chan error),
 		pollPeriod:        pollPeriod,
 		finalityDepth:     finalityDepth,
+		useFinalityTag:    useFinalityTag,
```

Comment:
hmm if a client is using finalityTag=true, I think they shouldn't have to specify finalityDepth at all and lp should just figure it out internally (right now it seems required to specify vaguely correct finalityDepth for the start block determination). So I'd probably validate that either finalityTag=true || finalityDepth > 0. We wouldn't want to just fail to boot if one RPC call fails though so can't just do a call to latestBlockAndFinalityDepth in the constructor/Start(). I think putting it the main log poller tick makes sense:
```
if no latest block in db {
	// first time poll
	latestBlock, finalityDepth, err := latestBlockAndFinalityDepth(ctx)
	// use latestBlock and finalityDepth to determine start block
} 	
```
Wdyt?

Reply:
>  So I'd probably validate that either (finalityTag=true && finalityDepth = 0) || (finalityDepth > 0 && finalityTag=false)

This is precisely what I suggested at some point, but then I realized that `finalityDepth` is used by more components (not only LogPoller). As long as it's a chain config that multiple components can read and use, enforcing such logic is tough. 

> I think putting it the main log poller tick makes sense

I would like to know if that will work. We might have a scenario in which
1. LogPoller starts with finality enabled
2. User asks for finalized logs (finality is not yet set)
3. LogPoller performs the first tick 

We can return errors from read queries using finality until LogPoller fetches this from chain. It will decrease system availability but will increase consistency. That might be a problem because the plugin will be able to process once RPC is back. For the CCIP case it might not be an issue because if RPC is dead, then CCIP won't work anyway because we are very chatty anyway. However, this might be the tradeoff that other products are not willing to take.